proto/frontend/src/pre_mc_mgr.cpp

```diff
@@ -147,13 +161,69 @@ struct PreMcMgr::NodeCleanupTask : public McLocalCleanupIface {
   pi_mc_node_handle_t node_h;
 };
 
+// Extracts the egress port of a given P4Runtime `replica` into the given
+// `ReplicaPort`.
+Status GetReplicaPort(const p4v1::Replica &replica, ReplicaPort &egress_port) {
+  egress_port.port_kind = replica.port_kind_case();
+  switch (replica.port_kind_case()) {
+    case p4v1::Replica::kEgressPort: {
+      egress_port.port_id = static_cast<pi_mc_port_t>(replica.egress_port());
+      RETURN_OK_STATUS();
+    }
+    case p4v1::Replica::kPort: {
+      egress_port.num_bytes = replica.port().size();
+      if (egress_port.num_bytes > sizeof(pi_mc_port_t)) {
+        RETURN_ERROR_STATUS(Code::UNIMPLEMENTED,
+                            "Multicast group replica `port` has %d bytes, but "
+                            "only %d bytes are supported",
+                            egress_port.num_bytes, sizeof(pi_mc_port_t));
+      }
+      egress_port.port_id = 0;
+      for (char byte : replica.port()) {
+        egress_port.port_id <<= 8;
+        egress_port.port_id += static_cast<uint8_t>(byte);
+      }
```

Comment:
could you reuse the existing function: https://github.com/p4lang/PI/blob/f043e6f5f4271076ad7e58aa9889c82dbfc8c3ca/proto/frontend/src/action_prof_mgr.cpp#L62. It probably makes sense to move it to common.h

Reply:
I moved this logic and the reverse conversion into common.h, but ended up modifying it a bit to make it more robust & reusable.